eth/beacon/state_machines/base.py

```diff
@@ -303,17 +304,18 @@ def process_block(
         )
 
         # Process per cycle state changes (CrystallizedState and ActiveState)
-        processed_crystallized_state, processed_active_state = cls.compute_cycle_transitions(
+        processed_crystallized_state, processed_active_state = self.compute_cycle_transitions(
             crystallized_state,
             processing_active_state,
+            block,
+            config,
```

Comment:
Making note that passing the `config` object around like this seems like an API that could be improved.  IIUC the `BeaconConfig` should be both immutable and static within the context of a single fork.  For this reason, does it make sense to make this a class property?

I'm starting to like this pattern and am considering what it would look like to port it over to the main `Chain` and `VM` APIs.  Many of the `constants` would become config properties which would ideally reduce some of the reliance on inheritance by giving each `VM` it's own `config` object (think replacing `VM.get_block_reward()` with `VM.config.block_reward`.